jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/ConfigureLambdaDialog.kt

```diff
@@ -0,0 +1,142 @@
+    private val project: Project,
+    private val queue: Queue
+) : DialogWrapper(project), CoroutineScope by ApplicationThreadPoolScope("ConfigureLambda") {
+    private val lambdaClient: LambdaClient = project.awsClient()
+    private val iamClient: IamClient = project.awsClient()
+    val view = ConfigureLambdaPanel(project)
+
+    init {
+        title = message("sqs.configure.lambda")
+        setOKButtonText(message("general.configure_button"))
+        setOKButtonTooltip(message("sqs.configure.lambda.configure.tooltip"))
+
+        init()
+    }
+
+    override fun createCenterPanel(): JComponent? = view.component
+
+    override fun getPreferredFocusedComponent(): JComponent? = view.lambdaFunction
+
+    override fun doValidate(): ValidationInfo? {
+        if (functionSelected().isEmpty()) {
+            return ValidationInfo(message("sqs.configure.lambda.validation.function"), view.lambdaFunction)
+        }
+        return null
+    }
+
+    override fun doOKAction() {
+        if (!isOKActionEnabled) {
+            return
+        }
+
+        isOKActionEnabled = false
+        setOKButtonText(message("sqs.configure.lambda.in_progress"))
+
+        launch {
+            try {
+                configureLambda(functionSelected())
+                runInEdt(ModalityState.any()) {
+                    close(OK_EXIT_CODE)
+                }
+                notifyInfo(message("sqs.service_name"), message("sqs.configure.lambda.success", functionSelected()), project)
+            } catch (e: InvalidParameterValueException) {
+                // Exception thrown for invalid permission
+                runInEdt(ModalityState.any()) {
+                    if (ConfirmIamPolicyDialog(project, iamClient, lambdaClient, functionSelected(), queue, view.component).showAndGet()) {
+                        retryConfiguration(functionSelected())
+                    } else {
+                        setOKButtonText(message("general.configure_button"))
+                        isOKActionEnabled = true
+                    }
+                }
+            } catch (e: Exception) {
+                LOG.warn(e) { message("sqs.configure.lambda.error", queue.queueName, functionSelected()) }
+                setErrorText(e.message)
+                setOKButtonText(message("general.configure_button"))
+                isOKActionEnabled = true
+            }
+        }
+    }
+
+    private fun functionSelected(): String = view.lambdaFunction.selected()?.functionName() ?: ""
+
+    internal fun configureLambda(functionName: String) {
+        lambdaClient.createEventSourceMapping {
+            it.functionName(functionName)
+            it.eventSourceArn(queue.arn)
+        }
+    }
+
+    // It takes a few seconds for the role policy to update, so this function will attempt configuration for a duration of time until it succeeds.
+    internal suspend fun waitUntilConfigured(functionName: String): String {
+        var identifier: String = ""
+        waitUntil(
```

Comment:
This throws a `WaiterTimeoutException` when it times out, so we need to handle it somewhere (Maybe in here then return `null`)?